src/decorators.py

```diff
@@ -192,7 +194,7 @@ def __exit__(self, exc_type, exc_value, tb):
 
 class handle_error:
 
-    def __new__(cls, func=None, *, instance=None):
+    def __new__(cls, func: Optional[Callable[..., None]]=None, *, instance=None):
```

Comment:
func shouldn't be optional here; it defaulting to None is wrong (should have no default)